lib/cubits/database_connection/database_connection_cubit.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:bloc/bloc.dart';
+import 'package:coffeecard/base/strings.dart';
+import 'package:coffeecard/data/repositories/v1/app_config_repository.dart';
+import 'package:coffeecard/generated/api/coffeecard_api_v2.swagger.swagger.dart';
+import 'package:coffeecard/service_locator.dart';
+import 'package:equatable/equatable.dart';
+import 'package:flutter/material.dart';
+
+part 'database_connection_state.dart';
+
+class DatabaseConnectionCubit extends Cubit<DatabaseConnectionState> {
+  final AppConfigRepository _configRepository = sl.get<AppConfigRepository>();
+
+  DatabaseConnectionCubit()
+      : super(
+          const DatabaseConnectionState(
+            status: DatabaseConnectionStatus.unknown,
+          ),
+        ) {
+    getConfig();
+  }
+
+  Future<void> getConfig() async {
+    final AppConfig config = await _configRepository.getAppConfig();
```

Comment:
I think it would be cleaner if the repository just returned what you _actually_ needs i.e. the environment status, so we don't have to parse it here in the cubit.

Reply:
the 'AppConfig' DTO only consists of a single field: 'environmentType'. Are you suggesting we should return this field from the 'getAppConfig' method instead of the 'AppConfig'?

Reply:
As this time, I think that could make sense, yes